metricproviders/datadog/datadog.go

```diff
@@ -0,0 +1,205 @@
+	}
+	response, err := httpClient.Do(request)
+
+	if err != nil {
+		return metricutil.MarkMeasurementError(measurement, err)
+	}
+
+	value, status, err := p.parseResponse(metric, response)
+	if err != nil {
+		return metricutil.MarkMeasurementError(measurement, err)
+	}
+
+	measurement.Value = value
+	measurement.Phase = status
+	finishedTime := metav1.Now()
+	measurement.FinishedAt = &finishedTime
+
+	return measurement
+}
+
+func (p *Provider) parseResponse(metric v1alpha1.Metric, response *http.Response) (string, v1alpha1.AnalysisPhase, error) {
+
+	bodyBytes, err := ioutil.ReadAll(response.Body)
+
+	if err != nil {
+		return "", v1alpha1.AnalysisPhaseError, fmt.Errorf("Received no bytes in response: %v", err)
+	}
+
+	if response.StatusCode == http.StatusForbidden || response.StatusCode == http.StatusUnauthorized {
+		return "", v1alpha1.AnalysisPhaseError, fmt.Errorf("received authentication error response code: %v %s", response.StatusCode, string(bodyBytes))
+	} else if response.StatusCode != http.StatusOK {
+		return "", v1alpha1.AnalysisPhaseError, fmt.Errorf("received non 2xx response code: %v %s", response.StatusCode, string(bodyBytes))
+	}
+
+	var res datadogResponse
+	err = json.Unmarshal(bodyBytes, &res)
+	if err != nil {
+		return "", v1alpha1.AnalysisPhaseError, fmt.Errorf("Could not parse JSON body: %v", err)
+	}
+
+	if len(res.Series) < 1 || len(res.Series[0].Pointlist) < 1 {
+		return "", v1alpha1.AnalysisPhaseError, fmt.Errorf("Datadog returned no value: %s", string(bodyBytes))
+	}
+
+	series := res.Series[0]
+	datapoint := series.Pointlist[len(series.Pointlist)-1]
+	if len(datapoint) < 1 {
+		return "", v1alpha1.AnalysisPhaseError, fmt.Errorf("Datadog returned no value: %s", string(bodyBytes))
+	}
+
+	status := evaluate.EvaluateResult(datapoint[1], metric, p.logCtx)
+	return strconv.FormatFloat(datapoint[1], 'f', -1, 64), status, nil
+}
+
+// Resume should not be used the Datadog provider since all the work should occur in the Run method
+func (p *Provider) Resume(run *v1alpha1.AnalysisRun, metric v1alpha1.Metric, measurement v1alpha1.Measurement) v1alpha1.Measurement {
+	p.logCtx.Warn("Datadog provider should not execute the Resume method")
+	return measurement
+}
+
+// Terminate should not be used the Datadog provider since all the work should occur in the Run method
+func (p *Provider) Terminate(run *v1alpha1.AnalysisRun, metric v1alpha1.Metric, measurement v1alpha1.Measurement) v1alpha1.Measurement {
+	p.logCtx.Warn("Datadog provider should not execute the Terminate method")
+	return measurement
+}
+
+// GarbageCollect is a no-op for the Datadog provider
+func (p *Provider) GarbageCollect(run *v1alpha1.AnalysisRun, metric v1alpha1.Metric, limit int) error {
+	return nil
+}
+
+func NewDatadogProvider(logCtx log.Entry, kubeclientset kubernetes.Interface) (*Provider, error) {
+	ns := Namespace()
+	secret, err := kubeclientset.CoreV1().Secrets(ns).Get(DatadogTokensSecretName, metav1.GetOptions{})
```

Comment:
FYI I updated the k8s libraries from underneath you, which now require context objects passed into every call. You can just use context.TODO() for now.